index.js

```diff
@@ -209,9 +222,38 @@
           }
           return isPropertyDefined('name', config) ? (config.name + '.' + commandName) : commandName;
         };
-
+        
+        // Try to read route configuration and log warning if not possible
+        var routes = {};
+        if (readFromRoute) {
+          var $route = $injector.get('$route');
+          if (!$route) {
+            $log.warn('$route service is not available. Make sure you have included ng-route in your application dependencies.');
+          } else {
+            routes = $route.routes;
+          }
+        }
         var getUrl = function () {
-          var url = trackUrlParams ? $location.url() : $location.path();
+          // Using ngRoute provided tracking urls
+          var url = $location.url();
+          var trackUrl;
+          Object.keys(routes).forEach(function (key) {
+            var route = routes[key];
+            // Check if url matches this route
+            if (!("regexp" in route) || !route.regexp.test(url)) {
```

Comment:
The linter is going to freak out about the use of double-quotes. All strings need to be in single quotes.